generate_and_send_a_creative_email_using_openai_and_stability_ai/main.bal

```diff
@@ -0,0 +1,80 @@
+import ballerina/io;
+import ballerina/mime;
+import ballerinax/googleapis.gmail;
+import ballerinax/openai.text;
+import ballerinax/stabilityai;
+
+configurable string gmailToken = ?;
+configurable string openAIToken = ?;
+configurable string stabilityAIKey = ?;
+configurable string recipientEmail = ?;
+
+final text:Client openAIText = check new ({auth: {token: openAIToken}});
+final gmail:Client gmail = check new ({auth: {token: gmailToken}});
+final stabilityai:Client stabilityAI = check new ({authorization: stabilityAIKey});
+
+public function main(string topic) returns error? {
+    fork {
+        // Generate greeting text and design in parallel
+        worker greetingWorker returns string|error? {
+            text:CreateCompletionRequest textPrompt = {
+                prompt: string `Generate a creative poem on the topic ${topic}.`,
+                model: "text-davinci-003",
+                max_tokens: 1000
+            };
+            text:CreateCompletionResponse completionRes = check openAIText->/completions.post(textPrompt);
+            return completionRes.choices[0].text;
+        }
+
+        worker imageWorker returns byte[]|error? {
+            stabilityai:TextToImageRequestBody payload = {text_prompts: [{"text": topic, "weight": 1}]};
+            stabilityai:ImageRes listResult = check stabilityAI->/v1/generation/["stable-diffusion-v1"]/
+                text\-to\-image.post(payload);
+            string? imageBytesString = listResult.artifacts[0].'base64;
+            if imageBytesString is () {
+                return error("Image byte string is empty.");
+            }
+            byte[] imageBytes = imageBytesString.toBytes();
+            var base64Decode = mime:base64Decode(imageBytes);
+            if base64Decode !is byte[] {
+                return error("Error in decoding the image byte string.");
+            }
+            return base64Decode;
+        }
+    }
+
+    record {|
+        string|error? greetingWorker;
+        byte[]|error? imageWorker;
+    |} results = wait {greetingWorker, imageWorker};
+
+    string? poem = check results.greetingWorker;
+    byte[]? image = check results.imageWorker;
+    if poem !is string || image !is byte[] {
+        return error("Error while generating the poem and the image.");
+    }
+
+    io:Error? fileWrite = io:fileWriteBytes("./image.png", image);
+    if fileWrite is io:Error {
+        return error("Error while writing the image to a file.");
+    }
+
+    poem = poem.trim();
+    if poem !is string {
+        return error("Error while removing white space characters from the start and the end.");
+    }
+    string:RegExp r = re `\n`;
+    poem = r.replaceAll(poem, "<br>");
+    if poem !is string {
+        return error("Error while replacing linebreaks with <br>.");
+    }
```

Comment:
we may be able to prompt the model to give the poem in html format. shall we check and see whether that would be a better approach?

Reply:
A good suggestion. However doesn't seem to work everytime. I got all of the following responses. Therefore it is better to do manually.

I'm sorry, as an AI language model, I am unable to generate an HTML code for the poem as it requires a specific structure and styling. However, I can provide you with a plain text version of the poem:

<pre>
&lt;html&gt;
  &lt;head&gt;
    &lt;title&gt;A Lighthouse on a Cliff&lt;/title&gt;
  &lt;/head&gt;
  &lt;body&gt;
    &lt;h1&gt;A Lighthouse on a Cliff&lt;/h1&gt;
    &lt;p&gt;A lighthouse on a cliff stands tall and proud,&lt;/p&gt;
    &lt;p&gt;Guiding ships safely through the fog and cloud.&lt;/p&gt;
    &lt;p&gt;Its beacon shines bright for all to see,&lt;/p&gt;
    &lt;p&gt;A symbol of hope and security.&lt;/p&gt;
    &lt;p&gt;The waves crash below with all their might,&lt;/p&gt;
    &lt;p&gt;But the lighthouse stands firm, a constant light.&lt;/p&gt;
    &lt;p&gt;Its walls have weathered many a storm,&lt;/p&gt;
    &lt;p&gt;But it still stands strong, a beacon warm.&lt;/p&gt;
    &lt;p&gt;So when the seas are rough and the night is dark,&lt;/p&gt;
    &lt;p&gt;Look to the lighthouse, a shining mark.&lt;/p&gt;
    &lt;p&gt;It will guide you home, safe and sound,&lt;/p&gt;
    &lt;p&gt;A lighthouse on a cliff, always around.&lt;/p&gt;
  &lt;/body&gt;
&lt;/html&gt;
</pre>

<!DOCTYPE html>
<html>
  <head>
    <title>A Lighthouse on a Cliff</title>
  </head>
  <body>
    <h1>A Lighthouse on a Cliff</h1>
    <p>A lighthouse on a cliff stands tall and proud,</p>
    <p>Guiding ships safely through the fog and cloud.</p>
    <p>Its beacon shines bright for all to see,</p>
    <p>A symbol of hope and security.</p>

    <p>The waves crash below with all their might,</p>
    <p>But the lighthouse stands firm, a constant light.</p>
    <p>Its walls have weathered many a storm,</p>
    <p>But it still stands strong, a beacon warm.</p>

    <p>So when the seas are rough and the night is dark,</p>
    <p>Look to the lighthouse, a shining mark.</p>
    <p>It will guide you home, safe and sound,</p>
    <p>A lighthouse on a cliff, always around.</p>
  </body>
</html>